networkit/cpp/centrality/GroupClosenessLocalSwaps.cpp

```diff
@@ -0,0 +1,362 @@
+        q.push(idx.first);
+        farness[idx.second] = 1;
+        distance[idx.first] = 0;
+        visited[idx.first] = true;
+    }
+
+    do {
+        const auto u = q.front();
+        q.pop();
+
+        // Cannot swap with nodes in the group
+        if (distance[u] == 0)
+            canSwap[idxMap.at(u)] = 0;
+
+        bool uIsLeaf = false;
+
+        G->forNeighborsOf(u, [&](const node v) {
+            // Whether v is in \Gamma_u i.e., the shortest path from S to v is realized only
+            // by u.
+            bool inGamma = true;
+
+            // Whether the node in the group that realizes the shortest distance to v has
+            // been found.
+            bool nearestNodeFound = false;
+
+            // Index of the node in the group that realizes the shortest distance to v.
+            index groupIdx = none;
+
+            if (!visited[v]) {
+                uIsLeaf = false;
+                distance[v] = distance[u] + 1;
+                visited[v] = true;
+                q.push(v);
+
+                for (size_t i = 0; i < group.size(); ++i) {
+                    const auto curGamma = gamma[group.size() * u + i];
+                    if (curGamma) {
+                        if (!nearestNodeFound) {
+                            nearestNodeFound = true;
+                            groupIdx = i;
+                        } else
+                            inGamma = false;
+                    }
+
+                    gamma[group.size() * v + i] = curGamma;
+                }
+
+                if (inGamma)
+                    ++farness[groupIdx];
+
+            } else if (distance[u] + 1 == distance[v]) {
+                inGamma = true;
+                nearestNodeFound = false;
+                bool subtract = false;
+
+                for (size_t i = 0; i < group.size(); ++i) {
+                    if (gamma[group.size() * v + i]) {
+                        if (!nearestNodeFound) {
+                            nearestNodeFound = true;
+                            groupIdx = i;
+                        } else {
+                            inGamma = false;
+                            break;
+                        }
+                    } else if (gamma[group.size() * u + i]) {
+                        gamma[group.size() * v + i] = 1;
+                        subtract = true;
+                    }
+                }
+                if (inGamma && subtract)
+                    --farness[groupIdx];
+            }
+
+            if (!distance[u] && !distance[v])
+                canSwap[idxMap[u]] = true;
+        });
+
+        if (distance[u] != 0 && (!uIsLeaf || distance[u] == count{1}))
+            stack[stackSize++] = u;
+
+    } while (!q.empty());
+}
+
+int64_t GroupClosenessLocalSwaps::computeFarnessDecrement(node v) {
+    std::fill(visited.begin(), visited.end(), false);
+
+    std::queue<node> q;
+    q.push(v);
+    distance[v] = 0;
+    visited[v] = true;
+
+    int16_t decrement{1};
```

Comment:
uint16_t seems to be a mistake here (?).

Reply:
Yes, that should be `int64_t`.